compile/compile.go

```diff
@@ -908,6 +967,8 @@ func (o *optimizer) Do(ctx context.Context) error {
 	// because otherwise the optimization outputs (e.g., support rules) would have to
 	// merged somehow. Instead of dealing with that, just run the optimizations in the
 	// order the user supplied the entrypoints in.
+
+	var flattenedAnnotations []*ast.AnnotationsRef
```

Comment:
We should probably check the optimization level maybe and only do this for `O=1`?

Reply:
Should be ok to do this w/o any explicit check.